File: src/main/java/leets/weeth/domain/user/application/mapper/CardinalMapper.java
package leets.weeth.domain.user.application.mapper;

import leets.weeth.domain.user.application.dto.request.CardinalSaveRequest;
import leets.weeth.domain.user.application.dto.response.CardinalResponse;
import leets.weeth.domain.user.application.dto.response.UserCardinalDto;
import leets.weeth.domain.user.domain.entity.Cardinal;
import leets.weeth.domain.user.domain.entity.User;
import leets.weeth.domain.user.domain.entity.UserCardinal;
import org.mapstruct.Mapper;
import org.mapstruct.MappingConstants;
import org.mapstruct.ReportingPolicy;

import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface CardinalMapper {

    Cardinal from(CardinalSaveRequest dto);

    CardinalResponse to(Cardinal cardinal);

    UserCardinalDto toUserCardinalDto(User user, List<UserCardinal> cardinals);
}

File: src/main/kotlin/com/weeth/domain/cardinal/application/mapper/CardinalMapper.kt
package com.weeth.domain.cardinal.application.mapper

import com.weeth.domain.cardinal.application.dto.request.CardinalSaveRequest
import com.weeth.domain.cardinal.application.dto.response.CardinalResponse
import com.weeth.domain.cardinal.domain.entity.Cardinal
import com.weeth.domain.club.domain.entity.Club
import org.springframework.stereotype.Component

@Component
class CardinalMapper {
    fun toEntity(
        club: Club,
        request: CardinalSaveRequest,
    ): Cardinal =
        Cardinal.create(
            club = club,
            cardinalNumber = request.cardinalNumber,
        )

    fun toResponse(cardinal: Cardinal): CardinalResponse =
        CardinalResponse(
            cardinal.id,
            cardinal.cardinalNumber,
            cardinal.status,
            cardinal.createdAt,
            cardinal.modifiedAt,
        )
}
